src/sonic_pc/mod.rs

```diff
@@ -180,8 +181,19 @@ impl<E: PairingEngine> PolynomialCommitment<E::Fr> for SonicKZG10<E> {
                 ));
 
                 let shifted_powers_of_g = pp.powers_of_g[lowest_shift_degree..].to_vec();
-                let shifted_powers_of_gamma_g =
-                    pp.powers_of_gamma_g[lowest_shift_degree..].to_vec();
+                let mut shifted_powers_of_gamma_g = BTreeMap::new();
+                // Also add degree 0.
+                for degree_bound in enforced_degree_bounds {
+                    let shift_degree = max_degree - degree_bound;
+                    let mut powers_for_degree_bound = vec![];
+                    for i in 0..=supported_hiding_bound + 1 {
```

Comment:
Just for clarity

```suggestion
                    for i in 0..=(supported_hiding_bound + 1) {
```

Reply:
OK, I wondered why you adopted this style!